server/routes/serviceProviderReferrals/serviceProviderReferralsController.ts

```diff
@@ -791,15 +798,31 @@ export default class ServiceProviderReferralsController {
 
     const endOfServiceReport = await this.interventionsService.getEndOfServiceReport(accessToken, req.params.id)
     const referral = await this.interventionsService.getSentReferral(accessToken, endOfServiceReport.referralId)
-    const serviceCategory = await this.interventionsService.getServiceCategory(
+    const serviceCategories = await this.findSelectedServiceCategories(
       accessToken,
-      referral.referral.serviceCategoryId
+      referral.referral.interventionId,
+      referral.referral.serviceCategoryIds
     )
     const serviceUser = await this.communityApiService.getServiceUserByCRN(referral.referral.serviceUser.crn)
 
-    const presenter = new EndOfServiceReportConfirmationPresenter(referral, serviceCategory)
+    const presenter = new EndOfServiceReportConfirmationPresenter(referral, serviceCategories[0])
     const view = new EndOfServiceReportConfirmationView(presenter)
 
     ControllerUtils.renderWithLayout(res, view, serviceUser)
   }
+
+  private async findSelectedServiceCategories(
```

Comment:
I wonder if this is something that could go on a `referralDecorator` object (like the `interventionDecorator` that Lawrence introduced recently) that looks through its serviceCatgegoryIds`

Reply:
yes, I left a comment on Lawrence's PR with exactly that suggestion! However, I think the class would need to be renamed.

Reply:
However - he hasn't been able to merge in his PR just yet. Would it be okay for me to make that change on a future PR?

Reply:
I've merged mine now. I've actually already written a method that does the same thing as this, on the branch that I'm working on. But let's not block this PR - once I've got that method merged then I'll do a scan through the codebase for things that can be updated.

Reply:
Yep, let's get this in